Authenticator.spec.ts

```diff
@@ -0,0 +1,154 @@
+	const publicKey =
+		"MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAzWKqXfw8HU6lMtoLdc1WEkEZP/Dmhx8JmfMMQxcUIiFUkObL9zAEo/pk4/1FCaAy/l14yX76OU97Eannq8FObjd8tU5UOqb4n9RpXO3md1JDIZkuqhjQwuCJax/0nyNY+WH5MLWCBgo5kw6R+AHBdCXQGSGoMGfm0qQAySDE1PmZWc/6sR4WacK+ooMO/YtP7HuQQeG8qsJ44wbQXaYlKupxJr3EDo+Un4N9/PHmXlTTz1u7/aO2KbzP+V6kevvPzf+mS+KvSLMYMgIuDiQXvlor9UeC1M5VNj7Trx6HKnoiekKUt3tL14cIHVKhpOTlN2l2yj4ImmAG3qZ4gMO6DwIDAQAB"
+	const issuer = userwidgets.User.Key.Issuer.create("jest", "all ages", publicKey, privateKey)
+	const authenticator = new pax2pay.Authenticator(publicKey)
+	it("one realm role", async () => {
+		const realm: pax2pay.Realm | undefined = undefined
+		const token = await issuer.sign({
+			name: { first: "", last: "" },
+			email: "",
+			role: JSON.stringify({ test: "finance" }),
+			permissions: "",
+		})
+		const authenticated = await authenticator.authenticate(token ?? "", {
+			[realm ?? "*"]: { treasury: { rebalance: true } },
+		})
+		console.log("authenticated: ", authenticated)
+		expect(authenticated).toBeTruthy()
+	})
+	it("several realms role example", async () => {
+		const realm: pax2pay.Realm | undefined = "test"
+		const token = await issuer.sign({
+			name: { first: "", last: "" },
+			email: "",
+			role: JSON.stringify({ test: "finance", testUK: "finance" }),
+			permissions: "",
+		})
+		const authenticated = await authenticator.authenticate(token ?? "", {
```

Comment:
This function should be able to handle undefined so we don't have to use '?? ""' everywhere.

Reply:
done